### PR TITLE
Added IndicNLP News Datasets for different Indian Regional Languages

### PR DESCRIPTION
**Pull request by Sakshamrzt**
- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### CODE REVIEW

File: mteb/tasks/Classification/guj/GujaratiNewsClassification.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from mteb.abstasks.AbsTaskClassification import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class GujaratiNewsClassification(AbsTaskClassification):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/guj/GujaratiNewsClassification.py:7]: This all seem to be the same underlying dataset so I would rephrase it as a Multilingual task.

File: mteb/tasks/Classification/guj/GujaratiNewsClassification.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        dataset={
+            "path": "Sakshamrzt/IndicNLP-Gujarati",
+            "revision": "b98dbe0d93464c1a4789db07e0702677239c770a",
+        },
+        description="A News classification dataset in Gujarati.",
+        reference="https://github.com/AI4Bharat/indicnlp_corpus/blob/master/ai4bharat-indicnlp-corpus-2020.pdf",
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["guj-Gujr"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/guj/GujaratiNewsClassification.py:19]: This would have to be rephrased to use a mapping: 
{"language_on_hf": ["guj-Gujr"], ...} 

When using a multilingual dataset.

File: mteb/tasks/Classification/multilingual/MassiveScenarioClassification.py [hunk 3]
```diff
@@ -48,7 +48,7 @@
     "sv": ["swe-Latn"],
     "sw": ["swa-Latn"],
     "ta": ["tam-Taml"],
-    "te": ["tel-Telu"],
+    "tel": ["tel-Telu"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/MassiveScenarioClassification.py:51]: ?

File: mteb/tasks/Classification/multilingual/MassiveIntentClassification.py [hunk 4]
```diff
@@ -48,7 +48,7 @@
     "sv": ["swe-Latn"],
     "sw": ["swa-Latn"],
     "ta": ["tam-Taml"],
-    "te": ["tel-Telu"],
+    "tel": ["tel-Telu"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/multilingual/MassiveIntentClassification.py:51]: ?

**Review by KennethEnevoldsen — COMMENTED** [review]
Thanks for the addition @Sakshamrzt, this seems like a really good dataset to add, but seems like we need to rephrase it as a multilingual task.

File: mteb/tasks/Classification/guj/GujaratiNewsClassification.py (hunk 1 above)
Comment (imenelydiaker) [reply mteb/tasks/Classification/guj/GujaratiNewsClassification.py:7]: I agree with @KennethEnevoldsen, it would be better to have only one task that is Multilingual. Since you created the datasets, mayeb you can put them all into one repo and add languages subsets as in [mlsum](https://huggingface.co/datasets/mlsum). Also, do you have any references for this dataset ?

Comment (Sakshamrzt) [reply mteb/tasks/Classification/guj/GujaratiNewsClassification.py:7]: Thanks a lot for the review. I will refactor this to be a multilingual task.
